src/shogun/mathematics/NormalDistribution.h

```diff
@@ -0,0 +1,152 @@
+#ifndef __NORMALDISTRIBUTION_H__
+#define __NORMALDISTRIBUTION_H__
+
+#include <shogun/lib/Lock.h>
+#include <shogun/lib/common.h>
+#include <shogun/lib/config.h>
+
+namespace shogun
+{
+	class NormalDistribution
```

Comment:
template by type 
```
template<typename T>
class NormalDistribution { 
NormalDistribution(T mean = 0, T stddev = 1);
...
};
```

Reply:
I'm not sure whether the algorithm would work with other types. I will check it.

Reply:
in that case let this be the specialization for T=float64_t